metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1664,13 +1664,14 @@ public CompletableFuture<UpdateFeaturesResponseData> updateFeatures(
     @Override
     public CompletableFuture<List<CreatePartitionsTopicResult>> createPartitions(
```

Comment:
You can handle the `validateOnly` here and downstream function calls do not have to know about it. This would greatly simplify the code because now `ReplicationControl` does not have to deal with parsing the result and handling the `validateOnly` flag

e.g. this function implementation would change to
```
        if (topics.isEmpty()) {
            return CompletableFuture.completedFuture(Collections.emptyList());
        }

        return appendWriteEvent("createPartitions", context.deadlineNs(), () -> {
            final ControllerResult<List<CreatePartitionsTopicResult>> result = replicationControl.createPartitions(topics);
            return validateOnly ? result.withoutRecords() : result;
        });
```